internal/stacksapi/client_test.go

```diff
@@ -0,0 +1,198 @@
+package stacksapi
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/buildkite/agent-stack-k8s/v2/internal/version"
+	"github.com/buildkite/roko"
+)
+
+func TestNewClient(t *testing.T) {
+	t.Parallel()
+	t.Run("creates client with defaults", func(t *testing.T) {
+		t.Parallel()
+		client, err := NewClient("test-token")
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		if client.baseURL.String() != DefaultBaseURL {
+			t.Errorf("expected baseURL %s, got %s", DefaultBaseURL, client.baseURL)
```

Comment:
I remain steadfast in advocating for "got before want" https://google.github.io/styleguide/go/decisions#got-before-want

Also `%q` is the superior choice here

```suggestion
			t.Errorf("client.baseURL = %q, want %q", client.baseURL, DefaultBaseURL)
```